java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -1742,6 +1745,52 @@ public static Table convertFromRows(ColumnVector vec, DType ... schema) {
     return new Table(convertFromRows(vec.getNativeView(), types, scale));
   }
 
+  /**
+   * Construct a table from a packed representation.
+   * @param metadata host-based metadata for the table
```

Comment:
Do we want to mention about the metadata buffer being rewound and possible read again? or should we reset the read/write pointers to where they were before this was called?

Reply:
Yeah that's a really good point.  Thinking about this a bit more, I believe it'll be safest if we return a new read-only instance of the direct byte buffer (which is relatively cheap to instantiate).  Besides the rewind issue, that should also fix any multithreading races where two threads try to read the buffer simultaneously.